misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java

```diff
@@ -89,6 +90,13 @@ protected JsonElement transformSyntaxNode(Node node) {
         }
         nodeJson.addProperty("source", node.toSourceCode());
         nodeJson.addProperty("kind", prettifyKind(node.kind().toString()));
+        // TODO: Generalize the diagnostic implementation.
```

Comment:
@NipunaMarcus Can you check whether you have any concerns regarding these changes?

Reply:
sure give me few minutes

Reply:
All seem good guys... BTW nicely done.